### DISCUSSION TITLE
full list of parameters that we'll need in the yaml files.

### DISCUSSION THREAD

**Discussion by bnord** [deepskies/DeepSZSim#58] [Ideas]
We'll need to select the list of parameters that we want (for us and the user) to be able to change.

Currently, we have
1. base config file that no one will want to change, but may be needed: https://github.com/deepskies/SimSZ/blob/dev-pipeline/SimSZ/settings/base_config.yaml
2. general config file that people will want to change: https://github.com/deepskies/SimSZ/blob/dev-pipeline/SimSZ/settings/config.yaml

In this discussion, we can brainstorm which parameters should be included in any release, and we should mark ones that we want to for the alpha release.

**Comment by bnord** [comment]
# Battaglia profile parameters

I imagine that some of the Battaglia profile parameters are or can be associated with systematic physics effects, and that we may want to change those profile parameters in a simulation? Or perhaps allow the parameters to come from a distribution?

**Reply by evavagiakis** [reply]
The Battaglia profile is only defined using z, M, R and a cosmology

**Reply by bnord** [reply]
R is radius?

Assuming so, will the choice of R correspond to the mass? e.g., 200 vs. Vir?

**Reply by evavagiakis** [reply]
That's right, and R comes about from a colossus call, so z, M and a cosmology are the original inputs (editing what I said above)

**Comment by evavagiakis** [comment]
dev-massdist is the only branch being updated this year, please look at the .yamls there

**Comment by bnord** [comment]
# Current yaml files for configuration

1. There are two yaml files to consider for what users may want to update.
2. these have overlaps
3. (There should probably only be one yaml file, and if there is more than one yaml file, they shouldn't overlap in parameters)

https://github.com/deepskies/SimSZ/blob/dev-massdist/inputdata.yaml
COSMOLOGY:
  Planck2018: #https://arxiv.org/abs/1807.06209:
    flat: True 
    H0: 67.4
    Omega_m0: 0.315
    Omega_b0: 0.049
    Omega_d0: 0.636
    sigma8: 0.83
    ns: 0.965
    cosmo_h: 0.674
    t_cmb: 2.725 #K

SURVEYS:
  ACTDR5:
    150: #Observation frequency [GHz]
      beam_size: 1.3 #From published survey properties
      noise_level: 30 #From published survey properties
  
IMAGES:
  image_size: 37
  pixel_scale: 0.5
    
    
    
    
 https://github.com/deepskies/SimSZ/blob/dev-massdist/config.yaml
COSMOLOGY:
  Planck2018: #https://arxiv.org/abs/1807.06209:
    flat: True 
    H0: 67.4
    Omega_m0: 0.315
    Omega_b0: 0.049
    Omega_d0: 0.636
    sigma8: 0.83
    ns: 0.965
    cosmo_h: 0.674
    t_cmb: 2.725 #K
  Battaglia2012:
    flat: True 
    H0: 70
    Omega_m0: 0.25
    Omega_b0: 0.043
    sigma8: 0.8
    ns: 0.96
    cosmo_h: 0.7
    t_cmb: 2.725 #K

SURVEYS:
  ACTDR4: #https://arxiv.org/pdf/2007.07290
    90: #Observation frequency [GHz]
      beam_size: 2.1 #From published survey properties 
      noise_level: 100 #From published survey properties 
    150: #Observation frequency [GHz]
      beam_size: 1.4 #From published survey properties 
      noise_level: 75 #From published survey properties 
    220: #Observation frequency [GHz]
      beam_size: 1.05 #From published survey properties
      noise_level: 41 #From published survey properties
  ACTDR5:
    90: #Observation frequency [GHz]
      beam_size: 2.1 #From published survey properties
      noise_level: 40 #From published survey properties
    150: #Observation frequency [GHz]
      beam_size: 1.3 #From published survey properties
      noise_level: 30 #From published survey properties
    220: #Observation frequency [GHz]
      beam_size: 1.0 #From published survey properties
      noise_level: 125 #From published survey properties
  SPT: #https://arxiv.org/pdf/1111.7245
    150: #Observation frequency [GHz]
      beam_size: 1.15 #From published survey properties
      noise_level: 17 #From published survey properties
  
IMAGES:
  image_size: 37
  pixel_scale: 0.5

**Reply by bnord** [reply]
There's a function that appears to call/use a pressure profile. Is this used in the Battaglia profile construction? If so, the parameters in the pressure profile definition should potentially be considered as inputs to the pipeline.

**Comment by evavagiakis** [comment]
I believe there are two .yamls here because during the retreat it was said there should be one main .yaml which contains all the possible inputs, and then a run-specific .yaml which contains only the information needed for a specific run. In this directory, https://github.com/deepskies/SimSZ/blob/dev-massdist/config.yaml was the main .yaml and https://github.com/deepskies/SimSZ/blob/dev-massdist/inputdata.yaml was the run-specific .yaml used in the notebook. 

The functions that create the pressure profile take M, z to create the pressure profile, and these are not in the .yamls

**Reply by bnord** [reply]
1. one of these would be an example
2. the other would be a comprehensive one with (nearly) all the possible options that we allow.

**Reply by bnord** [reply]
Do the pressure profile parameters (exponents, normalizations) never change?

**Reply by evavagiakis** [reply]
Sounds good, then inputdata.yaml would serve as the example. As we are currently working with this, the pressure profile fit parameters don't change since they are drawn from a specific fit

**Reply by bnord** [reply]
coolio. thanks. 

May I rename inputdata.yaml to example.yaml so we keep track of that?

**Reply by evavagiakis** [reply]
Sounds good to me!
